synthea/qpp_qicore.yaml

```diff
@@ -0,0 +1,156 @@
+          - location: ServiceRequest.intent
+            value: order
+          - location: ServiceRequest.encounter.reference
+            value: $getField([Procedure.encounter.reference])
+          - location: ServiceRequest.subject.reference
+            value: $findRef([Patient])
+          - location: ServiceRequest.status
+            value: completed # all procedures are exported as completed
+          - if: Procedure.performed.ofType(Period) # handle value setting for period choice type
+            location: ServiceRequest.authoredOn
+            value: $getField([Procedure.performed.start]) # period choice type
+          - if: Procedure.performed.ofType(dateTime) # handle value setting for datetime choice type
+            location: ServiceRequest.authoredOn
+            value: $getField([Procedure.performed]) #datetime
+          - location: ServiceRequest.code
+            value: $getField([Procedure.code])
+        writeback:
+          - location: Procedure.basedOn.reference
+            value: $setRef([ServiceRequest])
+      - resourceType: DeviceRequest
+        based_on:
+          resource: Device
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-devicerequest
+        fields:
+          - location: DeviceRequest.code.reference
+            value: $findRef([Device])
+          - location: DeviceRequest.authoredOn
+            value: $getField([Device.manufactureDate]) # manufacture time is set 3 weeks before device model's start, so this is close enough
+          - location: DeviceRequest.status
+            value: completed
+          - location: DeviceRequest.intent
+            value: order
+      # Creating one AdverseEvent per patient
+      - resourceType: AdverseEvent
+        based_on:
+          resource: Patient
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-adverseevent
+        fields:
+          - location: AdverseEvent.event.coding
+            value: $randomCode([http://hl7.org/fhir/ValueSet/adverse-event-type]) # may change in future to use a smaller subset of the ValueSet
+      # Creating one Communication per patient
+      - resourceType: Communication
+        based_on:
+          resource: Patient
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-communication
+        fields:
+          - location: Communication.status
+            value: $randomCode([http://hl7.org/fhir/ValueSet/event-status,code])
+      # Creating one MedicationAdministration for each MedicationRequest
+      - resourceType: MedicationAdministration
+        based_on:
+          resource: MedicationRequest
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-medicationadministration
+        fields:
+          - location: MedicationAdministration.status
+            value: completed
+          - location: MedicationAdministration.medication.coding
```

Comment:
The MedicationAdministration.medication should probably be set to the MedicationRequest.medication